backend/src/data/repositories/course/course.repository.ts

```diff
@@ -12,6 +15,15 @@ class Course {
     this.#CourseModel = CourseModel;
   }
 
+  async getAll(): Promise<(CourseM & { vendorKey: VendorKey })[]> {
+    return this.#CourseModel
+      .query()
+      .join('vendors', 'vendors.id', 'courses.vendorId')
+      .select('courses.*', 'vendors.key as vendorKey') as QueryBuilder<
```

Comment:
Let's provide `relationMappings` on vendors for course model and use `joinRelated` here